app/main.py

```diff
@@ -1,6 +1,19 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    results = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> int:
+        key = args + tuple(kwargs.items())
+        if key in results:
+            print("Getting from cache")
+            return results[key]
+        else:
```

Comment:
There is no need for `else` since the code will not execute after the `return` statement anyway. Please, remove the redundant `else`